proposed/middleware-meta.md

```diff
@@ -0,0 +1,133 @@
+PSR-N Meta Document
+===================
+
+1. Summary
+----------
+
+The purpose of this PSR is to provide an interface that defines the formal
+method signature of a PSR-7 compliant HTTP middleware component based on
+existing informal standards.
+
+2. Why Bother?
+--------------
+
+The general concept of a reusable request/response middleware was present in
+[StackPHP][stackphp]. Since the release of PSR-7, a number of frameworks have
+applied the StackPHP concept with a different interface signature. The signature
+for these middleware implementations has been almost universally the same.
+
+Agreeing on a formal middleware interface eliminates several problems and
+provides a number of benefits:
+
+* Provides a formal standard for middleware developers to commit to.
+* Eliminates duplication of similar interfaces defined by various frameworks.
+* Avoids minor discrepancies in method signatures.
+* Enables any middleware component to run in any compatible framework.
+
+Although many frameworks already use a similar signature, a formal PSR will
+help middleware and framework vendors to formally certify compliance.
+
+[stackphp]: http://stackphp.com/
+
+3. Scope
+--------
+
+## 3.1 Goals
+
+* Provide a middleware interface that is compatible with PSR-7 as a Composer package.
+* Implement an already widely-adopted informal standard.
+
+## 3.2 Non-Goals
+
+* Attempting to define the mechanism by which middleware is managed or dispatched.
+
+4. Approaches
+-------------
+
+### 4.1 Chosen Approach
+
+Based on the middleware implementations already used by frameworks that have
+adopted PSR-7, the following commonalities are observed:
+
+* The middleware is defined as a [callable][php-callable] using `__invoke`.
```

Comment:
Too bad that this RFC is still under discussion so middleware wouldn't need to be specifically tied to `__invoke`
https://wiki.php.net/rfc/callable-types